src/query/parser/promql/matchers.go

```diff
@@ -90,27 +90,35 @@ func NewAggregationOperator(expr *promql.AggregateExpr) (parser.Params, error) {
 		Without:      expr.Without,
 	}
 
-	op := getAggOpType(opType)
-	if op == common.UnknownOpType {
+	switch op := getAggOpType(opType); op {
+	case common.UnknownOpType:
 		return nil, fmt.Errorf("operator not supported: %s", opType)
-	}
 
-	if op == aggregation.BottomKType || op == aggregation.TopKType {
+	case aggregation.BottomKType, aggregation.TopKType:
 		val, err := resolveScalarArgument(expr.Param)
 		if err != nil {
 			return nil, err
 		}
 
 		nodeInformation.Parameter = val
 		return aggregation.NewTakeOp(op, nodeInformation)
-	}
 
-	if op == aggregation.CountValuesType {
+	case aggregation.CountValuesType:
 		nodeInformation.StringParameter = expr.Param.String()
 		return aggregation.NewCountValuesOp(op, nodeInformation)
-	}
 
-	return aggregation.NewAggregationOp(op, nodeInformation)
+	case aggregation.QuantileType:
+		val, err := resolveScalarArgument(expr.Param)
+		if err != nil {
+			return nil, err
+		}
+
+		nodeInformation.Parameter = val
```

Comment:
May be worth adding a test for this; may be a bit complex though since it would need to generate the op, get the node, then create a mock block and run it against functions (since we can't really inspect this otherwise; maybe add a `NodeInformation()` on `parser.Params`? Although that may lead to a deeper rabbithole)

Reply:
We've discussed this with @linasm and reached a similar conclusion about it being not very straightforward. For now, the tests in `aggregators.test` and `queries.json` should provide some coverage. I may look into this further in a follow-up PR.